### DISCUSSION TITLE
ConstantResolution: Should constants like ARGV be allowed?

### DISCUSSION THREAD

**Discussion by okuramasafumi** [rubocop/rubocop#11943] [Ideas]
I noticed that `Lint/ConstantResolution` cop treats `ARGV` as a constant and says we should fully qualify it. However:

```ruby
::SomeGem::CLI.run(::ARGV)
```

This looks ugly. The name `ARGV` comes from tradition and we don't want to treat it as a "constant" while technically it is. The same applies to `STDOUT`, for instance.

So I suggest that we should discuss the default "Ignore" option value for that cop.

**Comment by koic** [comment]
This can be resolved by configuring `Ignore: ['ARGV', 'STDOUT']`.

```yaml
Lint/ConstantResolution:
  Ignore:
     - ARGV
     - STDOUT
```

https://docs.rubocop.org/rubocop/1.66/cops_lint.html#lintconstantresolution
